### PR TITLE
Dhcp Relay Global

### PR DESCRIPTION
**Pull request by saichint**
This PR is for adding new provider for dhcp_relay_global.
Tests passed on all nexus platforms (including CI).

### CODE REVIEW

File: lib/cisco_node_utils/dhcp_relay_global.rb [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+module Cisco
+  # node_utils class for dhcp_relay_global
+  class DhcpRelayGlobal < NodeUtil
+    attr_reader :name
+
+    def initialize(name)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name == 'default'
+      @name = name.downcase
```
Comment (mikewiebe) [inline lib/cisco_node_utils/dhcp_relay_global.rb]: Why do you need to downcase name here?  You generate an `ArgumentError` above if the name does not equal 'default'.

**_UPDATE**_ Thinking about this some more, we don't even really need a name at all.  I would treat this like vtp where initialize just enables the feature if `instantiate=true`.

https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/vtp.rb#L29

File: lib/cisco_node_utils/cmd_ref/dhcp_relay_global.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  get_value: '/^ip dhcp relay information trust-all$/'
+  set_value: "<state> ip dhcp relay information trust-all"
+  default_value: false
+
+ipv4_relay:
+  kind: boolean
+  get_value: '/^ip dhcp relay$/'
+  set_value: "<state> ip dhcp relay"
+  N3k: &relay_default_true
+    auto_default: false
```
Comment (mikewiebe) [inline lib/cisco_node_utils/cmd_ref/dhcp_relay_global.yaml:44]: Why do you need to turn off auto_default here?

File: tests/test_dhcp_relay_global.rb [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+# of the DhcpRelayGlobal class.
+class TestDhcpRelayGlobal < CiscoTestCase
+  @skip_unless_supported = 'dhcp_relay_global'
+  DEFAULT_NAME = 'default'
+
+  # Tests
+
+  def setup
+    super
+    config_no_warn 'no feature dhcp'
```
Comment (mikewiebe) [inline tests/test_dhcp_relay_global.rb]: `config_no_warn('no feature dhcp')`

File: tests/test_dhcp_relay_global.rb [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    end
+    assert_equal(drg.default_ipv4_src_intf, drg.ipv4_src_intf)
+    drg.ipv4_src_intf = 'port-channel200'
+    assert_equal('port-channel200', drg.ipv4_src_intf)
+    drg.ipv4_src_intf = drg.default_ipv4_src_intf
+    assert_equal(drg.default_ipv4_src_intf, drg.ipv4_src_intf)
+  end
+
+  def test_ipv4_sub_option_circuit_id_custom
+    skip_nexus_i2_image?
```
Comment (mikewiebe) [inline tests/test_dhcp_relay_global.rb:181]: Is this property broken or unsupported on i2 images?

File: tests/test_dhcp_relay_global.rb [hunk 5]
```diff
@@ -26,16 +26,16 @@ class TestDhcpRelayGlobal < CiscoTestCase
...
 
   def teardown
     config_no_warn('no feature dhcp') if first_or_last_teardown
     super
   end
 
-  def create_dhcp_relay_global(name=DEFAULT_NAME)
-    DhcpRelayGlobal.new(name)
+  def create_dhcp_relay_global
+    DhcpRelayGlobal.new(true)
```
Comment (mikewiebe) [inline tests/test_dhcp_relay_global.rb]: Simplify `DhcpRelayGlobal.new`

File: tests/test_dhcp_relay_global.rb (hunk 4 above)
Comment (saichint) [reply tests/test_dhcp_relay_global.rb:181]: yes, this is unsupported only on i2. Just tested it

**Comment by saichint** [comment]
All minitests passed on all nxos platforms. Please merge if there are no more comments.

**Pull request merged**